keps/sig-network/4962-network-topology-standard/README.md

```diff
@@ -0,0 +1,938 @@
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This document proposes a standard for declaring cluster network topology in Kubernetes, representing the hierarchy of nodes, switches, and interconnects. In this context, a switch can refer to a physical network device or a collection of such devices with close proximity and functionality.
+
+## Motivation
+
+Understanding the cluster network topology is essential for optimizing the placement of workloads that require intensive inter-node communication. Currently, there is no standardized way to represent this information in Kubernetes, making it challenging to develop control plane components and applications that can leverage topology awareness.
+
+This information might be useful for various components and features, including:
+
+- Pod affinity sections in deployment and pod specs
+- Kueue topology-aware scheduling
+- Development of native scheduler plugins for topology-aware scheduling, for example:
+  - Topology-aware gang-scheduling scheduler plugin
+  - Gang-scheduling auto-scaler
+  - DRA scheduler plugin
+
+### Goals
+
+- Introduce a standard way of representing network topology in Kubernetes clusters
+
+### Non-Goals
+
+- Implement a topology-aware gang-scheduling scheduler plugin
+- Define or implement topology discovery mechanisms for CSPs or on-premises environments
+
+## Proposal
+
+We propose new node label and annotation types to capture network topology information:
+- Network Topology Label
+- Network QoS Annotation
+
+### Network Topology Label
+Format: `network.topology.kubernetes.io/<nw-switch-type>: <switch-name>`
+- `<nw-switch-type>`: Logical type of the network switch (can be one of the reserved names or a custom name)
+  - Reserved names: `accelerator`, `block`, `datacenter`, `zone`
+- `<switch-name>`: Unique identifier for the switch
+
+### Network QoS Annotation
+Format: `network.qos.kubernetes.io/switches: <QoS>`
+- `<QoS>`: A JSON object where each key is a switch name (matching the network topology label) with a value containing:
```

Comment:
So this object contains N items (of below structure), where N is the number of predefined topology units (accelerator, block, datacenter, zone), right?

What I want to ensure is that those needs to be changed/updated when the cluster grows/shrinks (i.e. that the don't define distance between nodes themselves). But given that for a given node its contents don't depend on other nodes (rather on placements in the physical network), this seems to be fine.

Reply:
@wojtek-t , that's correct, each node will contain QoS metrics for network switches on their path to the top of the topology graph.

Reply:
We simplified this proposal - removed custom labels and removed annotations.